lighthouse-core/test/gather/gatherers/source-maps-test.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env jest */
+
+jest.useFakeTimers();
+
+const Driver = require('../../../gather/driver.js');
+const Connection = require('../../../gather/connections/connection.js');
+const SourceMaps = require('../../../gather/gatherers/source-maps.js');
+const {createMockSendCommandFn, createMockOnFn} = require('../mock-commands.js');
+
+const mapJson = JSON.stringify({
+  version: 3,
+  file: 'out.js',
+  sourceRoot: '',
+  sources: ['foo.js', 'bar.js'],
+  names: ['src', 'maps', 'are', 'fun'],
+  mappings: 'AAgBC,SAAQ,CAAEA',
+});
+
+describe('SourceMaps gatherer', () => {
+  /**
+   * @param {{event: LH.Crdp.Debugger.ScriptParsedEvent, map: string, fetchError: string}} mapsAndEvents
+   * @return {Promise<LH.Artifacts['SourceMaps']>}
+   */
+  async function getResults(mapsAndEvents) {
+    const onMock = createMockOnFn();
+    for (const {event} of mapsAndEvents) {
+      onMock.mockEvent('Debugger.scriptParsed', event);
```

Comment:
suggested changes so
- it's only the connection being mocked
- some names changed

feel free to cherry pick anything you want :)
<details><summary>patchy patch</summary>

```
diff --git a/lighthouse-core/test/gather/gatherers/source-maps-test.js b/lighthouse-core/test/gather/gatherers/source-maps-test.js
index b2ab37cd..663fc2d2 100644
--- a/lighthouse-core/test/gather/gatherers/source-maps-test.js
+++ b/lighthouse-core/test/gather/gatherers/source-maps-test.js
@@ -25,20 +25,18 @@ const mapJson = JSON.stringify({
 
 describe('SourceMaps gatherer', () => {
   /**
-   * @param {{event: LH.Crdp.Debugger.ScriptParsedEvent, map: string, fetchError: string}} mapsAndEvents
+   * @param {Array<{event: LH.Crdp.Debugger.ScriptParsedEvent, map: string, fetchError: string}>} mapsAndEvents
    * @return {Promise<LH.Artifacts['SourceMaps']>}
    */
-  async function getResults(mapsAndEvents) {
+  async function runSourceMaps(mapsAndEvents) {
     const onMock = createMockOnFn();
-    for (const {event} of mapsAndEvents) {
-      onMock.mockEvent('Debugger.scriptParsed', event);
-    }
-
     const sendCommandMock = createMockSendCommandFn()
       .mockResponse('Debugger.enable', {})
       .mockResponse('Debugger.disable', {});
 
     for (const {map, event, fetchError} of mapsAndEvents) {
+      onMock.mockEvent('protocolevent', {method: 'Debugger.scriptParsed', params: event});
+
       if (event.sourceMapURL.startsWith('data:')) {
         // Only the source maps that need to be fetched use the `evaluateAsync` code path.
         continue;
@@ -53,9 +51,9 @@ describe('SourceMaps gatherer', () => {
     }
     const connectionStub = new Connection();
     connectionStub.sendCommand = sendCommandMock;
+    connectionStub.on = onMock;
 
     const driver = new Driver(connectionStub);
-    driver.on = onMock;
 
     const sourceMaps = new SourceMaps();
     await sourceMaps.beforePass({driver});
@@ -68,7 +66,7 @@ describe('SourceMaps gatherer', () => {
   }
 
   it('ignores script with no source map url', async () => {
-    const result = await getResults([
+    const artifact = await runSourceMaps([
       {
         event: {
           url: 'http://www.example.com/script.js',
@@ -77,7 +75,7 @@ describe('SourceMaps gatherer', () => {
         map: null,
       },
     ]);
-    expect(result).toEqual([]);
+    expect(artifact).toEqual([]);
   });
 
   it('fetches map for script with source map url', async () => {
@@ -90,8 +88,8 @@ describe('SourceMaps gatherer', () => {
         map: mapJson,
       },
     ];
-    const result = await getResults(mapsAndEvents);
-    expect(result).toEqual([
+    const artifact = await runSourceMaps(mapsAndEvents);
+    expect(artifact).toEqual([
       {
         scriptUrl: mapsAndEvents[0].event.url,
         map: JSON.parse(mapsAndEvents[0].map),
@@ -109,8 +107,8 @@ describe('SourceMaps gatherer', () => {
         fetchError: 'TypeError: Failed to fetch',
       },
     ];
-    const result = await getResults(mapsAndEvents);
-    expect(result).toEqual([
+    const artifact = await runSourceMaps(mapsAndEvents);
+    expect(artifact).toEqual([
       {
         scriptUrl: mapsAndEvents[0].event.url,
         errorMessage: 'TypeError: Failed to fetch',
@@ -135,8 +133,8 @@ describe('SourceMaps gatherer', () => {
         },
       },
     ];
-    const result = await getResults(mapsAndEvents);
-    expect(result).toEqual([
+    const artifact = await runSourceMaps(mapsAndEvents);
+    expect(artifact).toEqual([
       {
         scriptUrl: mapsAndEvents[0].event.url,
         errorMessage: 'SyntaxError: Unexpected token { in JSON at position 1',
```

</details>

Reply:
done, thanks